### ISSUE TITLE
Interval multiplication: bug report provided by @rklaehn

### ISSUE THREAD

**Issue by denisrosset** [non/spire#372]
```scala
object RationalTest3 extends App {
  val a = Interval(-1, 1)
  val b = Interval.above(1)
  val x = -1
  val y = 10
  println(s"$a * $b = ${a*b}")
  println(a.contains(x))
  println(b.contains(y))
  println((a*b).contains(x*y))
}
```

**Comment by denisrosset** [comment]
Actually, there are randomized tests in IntervalTest.scala, and the case above should have been exercised. I'm investigating why the current tests fail to verify it.

**Comment by denisrosset** [comment]
Ok, found it. The `Interval` arbitrary generator is using a `Double` to generate different interval types, thinking the `Double` is between 0 and 1. It is not, so the test cases are heavily biased towards bounded intervals.

**Comment by denisrosset** [comment]
Solved by #373 .
